src/main/java/net/fabricmc/loader/config/ConfigManagerImpl.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.config;
+
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.config.ConfigDefinition;
+import net.fabricmc.loader.api.config.serialization.ConfigSerializer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigEnvironment;
+import net.fabricmc.loader.api.config.entrypoint.ConfigPostInitializer;
+import net.fabricmc.loader.api.config.data.DataCollector;
+import net.fabricmc.loader.api.config.data.DataType;
+import net.fabricmc.loader.api.config.exceptions.ConfigSerializationException;
+import net.fabricmc.loader.api.config.util.ListView;
+import net.fabricmc.loader.api.config.value.ValueKey;
+import net.fabricmc.loader.api.config.value.ValueContainer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigInitializer;
+import net.fabricmc.loader.api.config.entrypoint.ConfigProvider;
+import net.fabricmc.loader.api.entrypoint.EntrypointContainer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConfigManagerImpl {
+	public static final Logger LOGGER = LogManager.getLogger("Fabric|Config");
+	private static final Map<ConfigDefinition<?>, List<ValueKey<?>>> CONFIGS = new HashMap<>();
+	private static final Map<ConfigDefinition<?>, ListView<ValueKey<?>>> CONFIG_VIEWS = new HashMap<>();
+	private static final Map<String, ConfigDefinition<?>> CONFIG_DEFINITIONS = new ConcurrentHashMap<>();
+	private static final Map<String, ValueKey<?>> CONFIG_VALUES = new ConcurrentHashMap<>();
+
+	private static ListView<ConfigDefinition<?>> CONFIG_DEFINITION_VIEW = null;
+
+	private static boolean FINISHED = false;
+
+	public static boolean isFinished() {
```

Comment:
All static methods make me cringe

Reply:
Using static methods is no worse than following the singleton pattern that Loader uses.

Reply:
A singleton is better in that you can port it more easily when you need to convert it into some subsystems. For example, you might have the loader load 2 distinct config subsystems for two sub-scenarios.